src/gromo/modules/growing_module.py

```diff
@@ -530,6 +531,25 @@ def __init__(
 
         self.layer: torch.nn.Module = layer.to(self.device)
         self.post_layer_function: torch.nn.Module = post_layer_function.to(self.device)
+        if extended_post_layer_function is None:
+            self.extended_post_layer_function = post_layer_function
```

Comment:
In case where `extended_post_layer_function` is None, the `post_layer_function` might trigger device related error if not pushed to the right device. Although this should not cause error due to PyTorch handling policy, the suggested modification provides a more robust implementation.
```suggestion
            self.extended_post_layer_function = post_layer_function.to(self.device)
```

Reply:
Thank you for pointing this out ! I believe we can also fix it by using `self.post_layer_function` which has already been moved to the correct device (two lines above). I will implement it soon